sqlmesh/core/snapshot/evaluator.py

```diff
@@ -918,6 +918,7 @@ def _promote_snapshot(
                 snapshots=snapshots,
                 deployability_index=deployability_index,
                 table_mapping=table_mapping,
+                runtime_stage=RuntimeStage.PROMOTING,
```

Comment:
Even though, on_virtual_update statements are defined in a separate block, having a dedicate runtime stage still makes sense for cases when the same macro is reused across multiple stages.